great_expectations/render/renderer_configuration.py

```diff
@@ -155,6 +154,66 @@ class CodeBlock(TypedDict):
     language: CodeBlockLanguage
 
 
+def make_exception_msg(v_type: RendererValueType, input: Any) -> str:
+    return f"Param type: <{v_type}> does not match value: <{input}>."
+
+
+def make_str_exception_msg(v_type: RendererValueType, input: Any) -> str:
+    message = f"Value was unable to be represented as a {v_type}"
+    try:
+        str(input)
+    except Exception as e:
+        message = f"{message}: {e}"
+    return message
+
+
+def safe_str(input: Any) -> bool:
+    try:
+        str(input)
+        return True
+    except Exception:
+        return False
+
+
+def safe_date(input: Any) -> bool:
+    try:
+        dateutil.parser.parse(input)
+        return True
+    except Exception:
+        return False
+
+
+PARAM_TYPE_TESTERS: Dict[
+    RendererValueType,
+    Tuple[
+        Callable[[Any], bool],
+        Callable[[RendererValueType, Any], str],
+    ],
+] = {
```

Comment:
My understanding is that since we dropped support for python 3.8, you should be able to use built-in types here (e.g. `dict` and `tuple`), but maybe I'm wrong about that. I do think this uses pydantic v1, which supports python 3.7.